pyodide_build/buildpkg.py

```diff
@@ -49,7 +51,16 @@ def download_and_extract(buildpath, packagedir, pkg, args):
         return srcpath
 
     if "url" in pkg["source"]:
-        tarballpath = buildpath / Path(pkg["source"]["url"]).name
+        response = request.urlopen(pkg["source"]["url"])
+        _, parameters = cgi.parse_header(
+            response.headers.get("Content-Disposition", "")
+        )
+        if "filename" in parameters:
+            tarballname = parameters["filename"]
+        else:
+            tarballname = Path(response.geturl()).name
+
+        tarballpath = buildpath / tarballname
```

Comment:
Since we are already making this HTTP call, can we then save the contents to `tarballpath` and avoid a second call with `wget` below?

Reply:
Good point, fixed. Thanks.